vignettes/r-developer-guide.Rmd

```diff
@@ -24,129 +24,178 @@ knitr::read_chunk(path = system.file("plumber", "loess", "plumber.R", package =
-* script: In the case of `plumbertableau`, this is a string that identifies the specific endpoint being requested.
-* data: This object contains the data being passed from Tableau to the extension. In the case of `plumbertableau`, this data is parsed and automatically made available to the underlying function. Note that values passed from Tableau are unnamed, and instead follow the naming convention `_arg1`, `_arg2`, ... ,`_argN`.
+![](files/swagger.png){width=75%}
 
-`plumbertableau` automatically provides a simple mock request that follows the specification provided by Tableau. This initial mock request contains simple `script` and `data` values. If you'd like, you can use `tableau_json()` to create a more full-featured request for use with Swagger:
+Click on the `/predict` endpoint to expand it. You can then click "Try it out" to modify the example request, and "Execute" to send it.
+
+![](files/swagger-predict.png){width=75%}
+
+Requests from Tableau consist of a JSON object with two items^[[Tableau Analytics Extensions API Reference](https://tableau.github.io/analytics-extensions-api/docs/ae_api_ref.html#post-evaluate)]. We can see these in the simple mock request that plumbertableau provides following Tableau's specification.
+
+```{json}
+{
+  "script": "/predict",
+  "data": {
+    "_arg1": [
+      0
+    ],
+    "_arg2": [
+      0
+    ]
+  }
+}
+```
+
+- `script`: A string identifying an endpoint. Behind the scenes, Tableau sends all extension requests to the `/evaluate` endpoint, and plumbertableau uses this to correctly route the request.
+- `data`: An object containing arguments for the function at the endpoint. plumbertableau parses this data and passes it to the R code. Tableau does not allow explicit naming of arguments, instead naming them `_arg1`, `_arg2`, ... ,`_argN`. Arguments are passed to R in the order in which they appear. In this case, `_arg1` will be passed to `x` and `_arg2` will be passed to `y`.
+
+To generate a more complex example, plumbertableau, we can use `tableau_json()` to create a more full-featured request for use with Swagger. Here, we pass in a data frame with two columns, and `tableau_json()` uses those columns for `_arg` and `_arg2`.
 
 ```{r}
 tableau_json(script = "/predict", 
              data = mtcars[,c("hp", "mpg")])
 ```
 
-Note that you will have to stop the running API in order to generate the mock request body in the current R session.
-
-Once you've generated a mock request body, you can insert it into the Swagger documentation for the API to test the API.
-
-![Swagger testing](files/swagger-test.png){width=75%}
+Note that because the running API occupies your R session, you'll have to stop running it to use `tableau_json()`. But once you've generated a mock request, you can paste it into the Swagger documentation to test the API. The data that would be sent back to Tableau appears under "Response body".
```

Comment:
```suggestion
Note that because the running API occupies your R session, you'll have to stop running it to use `tableau_json()`. Once you've generated a mock request, you can run the API again and paste the request into the Swagger documentation to test the API. The data that would be sent back to Tableau appears under "Response body".
```